### PR TITLE
Add brpop timeout option

### PR DESCRIPTION
**Pull request by dlunch**
Add brpop timeout option to reduce number of calls on redis

### CODE REVIEW

File: kombu/transport/redis_cluster.py [hunk 1]
```diff
@@ -221,7 +221,8 @@ def _register_BRPOP(self, channel):
                 except Exception as e:
                     logger.error('Error while registering BRPOP', extra={"e": e, "key": conn.key})
 
-        channel._brpop_start()
+        timeout = channel.connection.client.transport_options.get('brpop_timeout', 1)
```
Comment (bellatoris) [inline kombu/transport/redis_cluster.py:224]: what if we do not set the timeout at all? I believe it is the best way to do not use cpu

Comment (dlunch) [reply kombu/transport/redis_cluster.py:224]: It seems timeout value is mandatory https://redis.io/commands/brpop/

Comment (dlunch) [reply kombu/transport/redis_cluster.py:224]: We can set timeout to 0 to indefinitely block operation, https://redis.io/commands/blpop/ `A timeout of zero can be used to block indefinitely.`

Comment (dlunch) [reply kombu/transport/redis_cluster.py:224]: i'll test 0 timeout value after this pr merge

**Review by bellatoris — APPROVED** [review]

File: kombu/transport/redis_cluster.py (hunk 1 above)
Comment (bellatoris) [reply kombu/transport/redis_cluster.py:224]: Actually what I wonder is that there are any code blocks fall in problem if we set that timeout as zero

**Pull request merged**